### PR TITLE
Add GoOracleTags command to set build tags for oracle

### PR DESCRIPTION
**Pull request by gonotes**
When I reading docker source code, which locate in github.com/docker/docker/docker directory, I could not use GoReferrer to find "mainDaemon" when docker as daemon.What I found is client code when docker as client and without golang build tags "daemon".
So I try to add GoTags command,which can set oracle command param "-tags",then thing goes well!

### CODE REVIEW

**Comment by fatih** [comment]
Hi @gonotes 

Thanks for this addition. After that a couple of things: 

* Please remove the `godef` related section. It's totally unrelated to this PR.
* Please rename the command to `:GoOracleTags`. Build tags can be used in several different go commands, we need to distinguish between them.
* Add documentation too please (see the docs/ folder)

I also going to make comments to the PR, please fix them too.

File: autoload/go/oracle.vim [hunk 1]
```diff
@@ -115,6 +121,7 @@ func! s:RunOracle(mode, selected, needs_package) range abort
     " info check Oracle's User Manual section about scopes:
     " https://docs.google.com/document/d/1SLk36YRjjMgKqe490mSRzOPYEDe0Y_WQNRv-EiFYUyw/view#heading=h.nwso96pj07q8
     let cmd .= ' ' . go#util#Shelljoin(scopes)
+    " echon "vim-go: " | echohl Function | echon "current cmd is: '". cmd ."'" | echohl None
```
Comment (fatih) [inline autoload/go/oracle.vim]: Please remove this.

File: autoload/go/tool.vim [hunk 2]
```diff
@@ -4,6 +4,11 @@ function! go#tool#Files()
     else
         let command = "go list -f '{{range $f := .GoFiles}}{{$.Dir}}/{{$f}}{{printf \"\\n\"}}{{end}}{{range $f := .CgoFiles}}{{$.Dir}}/{{$f}}{{printf \"\\n\"}}{{end}}'"
     endif
+
+    if exists('g:go_oracle_tags')
+        let command .= ' ' .command 
+    endif
+
```
Comment (fatih) [inline autoload/go/tool.vim]: Please remove this.

File: ftplugin/go/commands.vim [hunk 3]
```diff
@@ -11,6 +11,7 @@ command! -range=% GoCallstack call go#oracle#Callstack(<count>)
 command! -range=% GoFreevars call go#oracle#Freevars(<count>)
 command! -range=% GoChannelPeers call go#oracle#ChannelPeers(<count>)
 command! -range=% GoReferrers call go#oracle#Referrers(<count>)
+command! -nargs=? GoTags call go#oracle#Tags(<f-args>)
```
Comment (fatih) [inline ftplugin/go/commands.vim]: As said previously, rename `GoTags` to `GoOracleTags`

File: autoload/go/oracle.vim [hunk 4]
```diff
@@ -96,18 +96,24 @@ func! s:RunOracle(mode, selected, needs_package) range abort
...
+        let tags = get(g:, 'go_oracle_tags')
+    else
+        let tags = ""
+    endif
 
     if a:selected != -1
         let pos1 = s:getpos(line("'<"), col("'<"))
         let pos2 = s:getpos(line("'>"), col("'>"))
-        let cmd = printf('%s -format plain -pos=%s:#%d,#%d %s',
+        let cmd = printf('%s -format plain -pos=%s:#%d,#%d -tags=%s %s',
```
Comment (fatih) [inline autoload/go/oracle.vim]: Did you test how oracle behaves if you pass an empty string `""` to the `-tags` flag?

File: autoload/go/oracle.vim (hunk 1 above)
Comment (gonotes) [inline autoload/go/oracle.vim]: Yeah,remove done.

File: autoload/go/tool.vim (hunk 2 above)
Comment (gonotes) [inline autoload/go/tool.vim]: Yeah,remove done.

File: ftplugin/go/commands.vim (hunk 3 above)
Comment (gonotes) [inline ftplugin/go/commands.vim]: Yeah, GoTags has rename to GoOracleTags.

File: autoload/go/oracle.vim (hunk 4 above)
Comment (gonotes) [inline autoload/go/oracle.vim]: Yes,when -tags is "",works fine.

**Comment by fatih** [comment]
@gonotes This is not mergable right now, fyi

**Comment by gonotes** [comment]
@fatih Conflit has been resolved.Many thanks for your help,this is my first time commit PR.

**Comment by fatih** [comment]
Thanks @gonotes. Just came back, sorry for the late response.

**Pull request merged**